drivers/spi/spi_max32.c

```diff
@@ -242,12 +274,52 @@ static int spi_max32_transceive(const struct device *dev)
 	dfs_shift = spi_max32_get_dfs_shift(ctx);
 
 	len = spi_context_max_continuous_chunk(ctx);
+
+#ifdef CONFIG_SPI_RTIO
+	switch (sqe->op) {
+	case RTIO_OP_RX:
+		len = sqe->rx.buf_len;
+		data->req.rxData = sqe->rx.buf;
+		data->req.rxLen = sqe->rx.buf_len;
+		data->req.txData = NULL;
+		data->req.txLen = len >> dfs_shift;
+		break;
+	case RTIO_OP_TX:
+		len = sqe->tx.buf_len;
+		data->req.rxData = NULL;
+		data->req.rxLen = 0;
+		data->req.rxData = data->dummy;
```

Comment:
I don't think this will work. Suggested change:
```suggestion
		data->req.txData = sqe->tx.buf;
```

Reply:
Hey @ubieda ,
That is problem which is fixed, it should point to tx.buf but i didn't update the PR since I was working on Sensor driver and I did some small optimizations but today/tomorrow i will update this PR with the latest code and it is detailed tested not using loopback but using real sensor data and the rtio transfers using interrupt are all async